src/main/java/org/truffleruby/extra/ffi/Pointer.java

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright (c) 2017 Oracle and/or its affiliates. All rights reserved. This
+ * code is released under a tri EPL/GPL/LGPL license. You can use it,
+ * redistribute it and/or modify it under the terms of the:
+ *
+ * Eclipse Public License version 1.0
+ * GNU General Public License version 2
+ * GNU Lesser General Public License version 2.1
+ */
+package org.truffleruby.extra.ffi;
+
+import com.oracle.truffle.api.CompilerDirectives.TruffleBoundary;
+import jnr.ffi.Runtime;
+import org.truffleruby.core.FinalizationService;
+import sun.misc.Unsafe;
+
+import java.lang.reflect.Field;
+import java.nio.ByteBuffer;
+import java.nio.charset.Charset;
+
+public class Pointer implements AutoCloseable {
+
+    public static final jnr.ffi.Pointer JNR_NULL = Runtime.getSystemRuntime().getMemoryManager().newOpaquePointer(0);
+
+    public static final Pointer NULL = new Pointer(0);
+
+    public static Pointer malloc(long size) {
+        return new Pointer(UNSAFE.allocateMemory(size));
+    }
+
+    public static Pointer calloc(long size) {
```

Comment:
Could you add some minimal JavaDoc to malloc/calloc to clearly mention whether they zero-out the allocated memory?
It will make it easier to use the right one when writing new code.